std/algorithm/searching.d

```diff
@@ -2330,6 +2330,31 @@ if (isForwardRange!R1 && isForwardRange!R2
     assert(findSkip(s, "def") && s.empty);
 }
 
+private struct FindSplitResult(R1, R2)
+    if (isForwardRange!R1 &&
+        isForwardRange!R2)
+{
+    this(R1 pre, R1 separator, R2 post)
+    {
+        asTuple = typeof(asTuple)(pre, separator, post);
+    }
+
+    Tuple!(R1, "pre", R1, "separator", R2, "post") asTuple;
```

Comment:
`Tuple!(R1, "pre", R1, "separator", R2, "post")` is in turn a subtype of `Tuple!(R1, R1, R2)`, so it should be backwards-compatible. The issue is that this subtyping uses a reinterpret cast which precludes CTFE.